crates/wasm/tests/test_build.rs

```diff
@@ -0,0 +1,469 @@
+            assets: "ASSETS".to_string(),
+            notes: "NOTES".to_string(),
+            spendable_notes: "SPENDABLE_NOTES".to_string(),
+            swaps: "SWAPS".to_string(),
+        };
+
+        let constants: IndexedDbConstants = IndexedDbConstants {
+            name: "penumbra-db-wasm-test".to_string(),
+            version: 1,
+            tables,
+        };
+
+        // Sample chain and fmd parameters.
+        let chain_params = ChainParameters {
+            chain_id: "penumbra-testnet-iapetus".to_string(),
+            epoch_duration: 5u64,
+        };
+
+        let fmd_params = FmdParameters {
+            precision_bits: 0u32,
+            as_of_block_height: 1u64,
+        };
+
+        // Serialize the parameters into `JsValue`.
+        let js_chain_params_value: JsValue = serde_wasm_bindgen::to_value(&chain_params).unwrap();
+        let js_fmd_params_value: JsValue = serde_wasm_bindgen::to_value(&fmd_params).unwrap();
+        let js_constants_params_value: JsValue = serde_wasm_bindgen::to_value(&constants).unwrap();
+
+        // Construct `WasmPlanner` instance.
+        let mut wasm_planner = WasmPlanner::new(
+            js_constants_params_value,
+            js_chain_params_value,
+            js_fmd_params_value,
+        )
+        .await
+        .unwrap();
+
+        // Create spendable UTXO note in JSON format.
+        let spendable_note_json = r#"
+        {
+            "note_commitment": {
+                "inner": "MY7PmcrH4fhjFOoMIKEdF+x9EUhZ9CS/CIfVco7Y5wU="
+            },
+            "note": {
+                "value": {
+                    "amount": {
+                        "lo": "1000000",
+                        "hi": "0"
+                    },
+                    "asset_id": {
+                        "inner": "nwPDkQq3OvLnBwGTD+nmv1Ifb2GEmFCgNHrU++9BsRE=",
+                        "alt_bech32m": "",
+                        "alt_base_denom": ""
+                    }
+                },
+                "rseed": "p2w4O1ognDJtKVqhHK2qsUbV+1AEM/gn58uWYQ5v3sM=",
+                "address": {
+                    "inner": "F6T1P51M1QOu8NGhKTMdJTy72TDhB2h00uvlIUcXVdovybq4ZcOwROB+1VE/ar4thEDNPanAcaYOrL+FugN8e19pvr93ZqmTjUdOLic+w+U=",
+                    "alt_bech32m": ""
+                }
+            },
+            "address_index": {
+                "account": "0",
+                "randomizer": "AAAAAAAAAAAAAAAA"
+            },
+            "nullifier": {
+                "inner": "8TvyFVKk16PHcOEAgl0QV4/92xdVpLdXI+zP87lBrQ8="
+            },
+            "height_created": "250305",
+            "height_spent": "0",
+            "position": "3204061134848",
+            "source": {
+                "inner": "oJ9Bo9v22srtUmKdTAMVwPOuGumWE2cAuBbZHci8B1I="
+            }
+        }
+        "#;
+
+        // Convert note to `SpendableNoteRecord`.
+        let spendable_note: SpendableNoteRecord =
+            serde_json::from_str(spendable_note_json).unwrap();
```

Comment:
I wonder if there's a way we can use serde_wasm_bindgen with JsValue here instead of serde_json 🤔 . Maybe it doesn't matter though.